### ISSUE TITLE
no-side-effects false positive

### ISSUE THREAD

**Issue by Windvis** [ember-cli/eslint-plugin-ember#351]
Hey!

Lodash's set function seems to trigger the `no-side-effects` rule. It seems that any function called `set` will trigger this rule though.

```javascript
import set from 'lodash.set';

//  ...
  computedValue: computed('someProperty', function () {
   let test = {};

    if (this.someProperty) {
      set(test, 'nested.prop', this.someProperty); // This triggers the no-side-effects linting rule
    }
    
    return test;
  }),
// ...
```

Simply renaming the import resolves the problem of course.

**Comment by rwjblue** [comment]
Definitely seems like a bug to me...

**Comment by franzliedke** [comment]
Another false positive:

~~~js
{
  computedValue: computed(function () {
   return foo => this.set('val', foo);
  })
}
~~~

Or am I missing something? This is a closure, and thus not a side effect...
